archive/frames/serializers.py

```diff
@@ -124,13 +136,7 @@ def create_or_update_header(self, frame, data):
         Headers.objects.update_or_create(defaults={'data': data}, frame=frame)
 
     def create_related_frames(self, frame, data):
-        related_frame_keys = [
-            'L1IDBIAS', 'L1IDDARK', 'L1IDFLAT', 'L1IDSHUT',
-            'L1IDMASK', 'L1IDFRNG', 'L1IDCAT', 'TARFILE',
-            'ORIGNAME', 'L1IDARC', 'L1IDTMPL', 'L1IDTRAC',
-        ]
-        for key in related_frame_keys:
-            related_frame = data.get(key)
+        for related_frame in data:
             if related_frame and related_frame != frame.basename:
```

Comment:
I see - so we must be delegating the responsibility to parsing the related frame keys to a client library?

Reply:
yes, posted above: https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/input/headerdata.py#L52.